Framework/PythonInterface/test/python/plugins/algorithms/FindUBFromScatteringPlaneTest.py

```diff
@@ -0,0 +1,171 @@
+import unittest
+from mantid.simpleapi import (
+    SetUB,
+    CreatePeaksWorkspace,
+    FindUBFromScatteringPlane,
+    LoadEmptyInstrument,
+    AddPeakHKL,
+    ClearUB,
+    IndexPeaks,
+    AnalysisDataService,
+)
+import numpy as np
+from unittest.mock import patch
+
+
+class FindUBFromScatteringPlaneTest(unittest.TestCase):
+    def setUp(self):
+        # load instrument + empty peakworkspace table
+        self.ws = LoadEmptyInstrument(Filename="SXD_Definition.xml", OutputWorkspace="empty_SXD")
+        axis = self.ws.getAxis(0)
+        axis.setUnit("TOF")
+        self.peaks1 = CreatePeaksWorkspace(InstrumentWorkspace=self.ws, NumberOfPeaks=0, OutputWorkspace="peaks1")
+
+    def assert_vector(self, ref_vector, calculated_vector, tol):
+        self.assertTrue(np.allclose(ref_vector, calculated_vector, atol=tol))
+
+    def index_peaks_helper(self, PeaksWorkspace, tolerance):
+        nindexed, avg_hkl_er, *_ = IndexPeaks(PeaksWorkspace, tolerance, RoundHKLs=False)
+        return nindexed
+
+    def up_helper(self, vertical_dir, set_vertical, Peaksworkspace):
+        lattice = Peaksworkspace.sample().getOrientedLattice()
+        res_vertical_angle = lattice.recAngle(*set_vertical, *vertical_dir)
+        self.assertLessEqual(np.sin(np.radians(res_vertical_angle)), np.sin(np.radians(5)))
+
+    def tearDown(self):
+        AnalysisDataService.clear()
+
+    def test_find_correct_ub_cubic(self):
+        peaks1 = self.peaks1
+        SetUB(peaks1, u=[1, -0.83, 0], v=[1, 0.83, 0], a=5.4, b=5.4, c=5.4, alpha=90, beta=90, gamma=90)
+        AddPeakHKL(peaks1, [2, 2, 0])
+        set_vertical = peaks1.getInstrument().getReferenceFrame().vecPointingUp()
+        ClearUB(peaks1)
+
+        FindUBFromScatteringPlane(
+            Vector1=[1, -1, 0], Vector2=[1, 1, 0], a=5.4, b=5.4, c=5.4, alpha=90, beta=90, gamma=90, PeaksWorkspace="peaks1"
+        )
+        vertical_dir = peaks1.getInstrument().getReferenceFrame().vecPointingUp()
```

Comment:
Think you could get the vertical direction in the `setUp` and store it as e.g. `self.vertical_dir`?